### PR TITLE
fix(#11530): markdown viewer enableSeeMoreVariant is not working as expected

### LINKED ISSUE #11530: markdown viewer enableSeeMoreVariant is not working as expected
**Affected module**
Does it impact the UI, backend or Ingestion Framework?
UI

**Describe the bug**
We are using richtext viewer to render the markdown and we have an option for read/less functionality which is based on the `enableSeeMoreVariant` flag, however, it is not working if we pass `false` to it and markdown values are getting clipped.

**To Reproduce**

Screenshots or steps to reproduce

**Expected behavior**

If `enableSeeMoreVariant` is false then we should show the full content.

**Version:**
 - Python version:
 - OpenMetadata version: [e.g. 0.8]
 - OpenMetadata Ingestion package version: [e.g. `openmetadata-ingestion[docker]==XYZ`]

**Additional context**

### PR DESCRIPTION
**Pull request by Sachin-chaurasiya**
### Describe your changes:

Fixes #11530 

I worked on issue #11530  and removed unused component `serviceRightPanel`

#
### Type of change:

- [x] Bug fix

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] My PR title is `Fixes <issue-number>: <short explanation>`
- [x] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

- [x] I have added a test that covers the exact scenario we are fixing. For complex issues, comment the issue number in the test for future reference.

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/components/common/rich-text-editor/RichTextEditorPreviewer.test.tsx [hunk 1]
```diff
@@ -344,4 +350,89 @@ describe('Test RichTextEditor Previewer Component', () => {
 
     expect(markdownParser).toBeInTheDocument();
   });
+
+  it('Should render read more button if enableSeeMoreVariant is true and max length is less than content length', () => {
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/common/rich-text-editor/RichTextEditorPreviewer.test.tsx:354]: Let's separate both cases
1. should render "more" if enableSeeMoreVariant is true
2. should render "more" if max length is less than content length

Also more inclined to follow above format to mention use-cases for more readable manner

Change "Read more" --> "More"
"Read less" --> "Less"

File: openmetadata-ui/src/main/resources/ui/src/components/common/rich-text-editor/RichTextEditorPreviewer.tsx [hunk 2]
```diff
@@ -32,13 +32,30 @@ const RichTextEditorPreviewer = ({
...
+
+  // initially read more will be false
+  const [readMore, setReadMore] = useState<boolean>(false);
+
+  // read more toggle handler
+  const handleReadMoreToggle = () => setReadMore((pre) => !pre);
+
+  // whether has read more content or not
+  const hasReadMore = useMemo(
+    () => enableSeeMoreVariant && markdown.length > maxLength,
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/common/rich-text-editor/RichTextEditorPreviewer.tsx:44]:
```suggestion
    () => enableSeeMoreVariant || markdown.length > maxLength,
```

Comment (Sachin-chaurasiya) [reply openmetadata-ui/src/main/resources/ui/src/components/common/rich-text-editor/RichTextEditorPreviewer.tsx:44]: @chirag-madlani , we have to check for both conditions here, as we only want to show the read more/less button if `enableSeeMoreVariant` is true and the markdown length is greater than the max length.

File: openmetadata-ui/src/main/resources/ui/src/components/common/rich-text-editor/RichTextEditorPreviewer.test.tsx (hunk 1 above)
Comment (Sachin-chaurasiya) [reply openmetadata-ui/src/main/resources/ui/src/components/common/rich-text-editor/RichTextEditorPreviewer.test.tsx:354]: Sure, I will change the use case text, however, we have separate cases if you look at the tests after that test.

**Review by chirag-madlani — APPROVED** [review]

**Pull request merged**